alica_engine/include/engine/containers/ConcurrentQueue.h

```diff
@@ -0,0 +1,3742 @@
+// Provides a C++11 implementation of a multi-producer, multi-consumer lock-free queue.
```

Comment:
we haven't decided to use this library yet. Even if we do, it's source code shouldn't be part of our repo. Rather we should pull or install it.